rust/types/src/api_types.rs

```diff
@@ -1656,6 +1695,8 @@ impl From<(KnnBatchResult, IncludeList)> for QueryResponse {
 pub enum QueryError {
     #[error("Error executing plan: {0}")]
     Executor(#[from] ExecutorError),
+    #[error("System time error: {0}")]
+    CouldNotGetTime(#[from] SystemTimeError),
```

Comment:
Nit - This error is probably in the realm of things we shouldn't error customers for, preferring instead to just not add this data to the latency payload of the metering event.